src/libclient/measurement/ping/ping_linux.cpp

```diff
@@ -252,10 +252,28 @@ Result Ping::result() const
     QVariantList roundTripMs;
     float avg = 0.0;
     QList<float> tmpPingTime = pingTime;
+    quint32 tmpPingsReceived = m_pingsReceived;
+    quint32 tmpPingsSent = m_pingsSent;
 
     // calculate average and fill ping times
     foreach (float val, tmpPingTime)
     {
+        // RTTs greater than receiveTimeout are discarded and don't influence the result
+        if (val >= definition->receiveTimeout)
+        {
+            if (tmpPingsReceived > 0)
```

Comment:
It would be better to do this whole thing when you write into pingTime (don't write it into pingTime and don't count this ping as received. While I looked at the code for this I realized the m_pingsReceived/sent stuff looks confusing in the TCP part of the code. Is it correct that you add/subtract these values multiple times?)